### PR TITLE
April fools funnies

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:
CL: [New] Oddly the assistantes seem more noisy then normal

### CODE REVIEW

**Comment by corp-0** [comment]
is there no way to add the new package as a dependency instead of directly adding the code to our codebase?

File: UnityProject/Assets/Scripts/Core/Input System/KeyboardInputManager.cs [hunk 1]
```diff
@@ -236,9 +236,10 @@ private bool CheckComboEvent(KeyCombo keyCombo, KeyEventType keyEventType = KeyE
 		{ KeyAction.ChatLocal,		() => { ChatUI.Instance.OpenChatWindow(ChatChannel.Local); }},
 		{ KeyAction.ChatRadio,		() => { ChatUI.Instance.OpenChatWindow(ChatChannel.Common); }},
 		{ KeyAction.ChatOOC,		() => { ChatUI.Instance.OpenChatWindow(ChatChannel.OOC); }},
-		{ KeyAction.ToggleHelp,    () => { ChatUI.Instance.OnHelpButton(); }},
+		{ KeyAction.ToggleHelp,     () => { ChatUI.Instance.OnHelpButton(); }},
 		{ KeyAction.ToggleAHelp,    () => { ChatUI.Instance.OnAdminHelpButton(); }},
 		{ KeyAction.ToggleMHelp,    () => { ChatUI.Instance.OnMentorHelpButton(); }},
+		{ KeyAction.PushToTalk,     () => { VoiceChatManager.Instance.ClientPushToTalkPressed = !VoiceChatManager.Instance.ClientPushToTalkPressed; }},
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Input System/KeyboardInputManager.cs:242]: in a real push to talk, you have to hold the key to talk. This sounds more like a toggle.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Input System/KeyboardInputManager.cs:242]: yeah,  sadly the current key bind manager isn't really set up for that, 
though there is UI to show when you microphone is active

**Comment by corp-0** [comment]
so I imagine this chat isn't spatial, right? It is all global

**Comment by Bod9001** [comment]
it's spatial

**Comment by Bod9001** [comment]
only thing that looks fiddly for code scan
```
Sandbox violation: Access to type not allowed: [UnityEngine.AudioModule]UnityEngine.Microphone in Assembly Adrenak.UniMic.Runtime
```
how should we handle microphone access with secure stuff?

man This is stupid, who thought this was a good idea to add to c# xD
```
Sandbox violation: Access to type not allowed: [mscorlib]System.Runtime.Serialization.Formatters.Binary.BinaryFormatter in Assembly Assets
BinaryFormatter classes  implement the IRemotingFormatter interface to support remote procedure calls (RPCs), 

"The BinaryFormatter type is dangerous and is not recommended for data processing. Applications should stop using BinaryFormatter as soon as possible, even if they believe the data they're processing to be trustworthy. BinaryFormatter is insecure and can't be made secure."
```
Guess I'll have to swap it out for mirrors custom reader writers,

**Comment by MaxIsJoe** [comment]
we already let our selves access things like the clipboard, I don't see why we should start worrying about microphone access.
though in the future, it would be nice if we can tell players on stationhub that a fork has the ability to access stuff like your microphone

If we really want to make this extra secure, simply do the same thing we do for safeurl but for microphones, and explicitly only allow audio listening when it is allowed by the user.

**Comment by Bod9001** [comment]
yeah, I'm just wondering what people's thought on microphone access

**Pull request merged**